x/ccv/provider/keeper/relay_test.go

```diff
@@ -0,0 +1,83 @@
+package keeper_test
+
+import (
+	"testing"
+
+	"github.com/golang/mock/gomock"
+
+	cryptocodec "github.com/cosmos/cosmos-sdk/crypto/codec"
+	ibcsimapp "github.com/cosmos/ibc-go/v3/testing/simapp"
+	testkeeper "github.com/cosmos/interchain-security/testutil/keeper"
+	ccv "github.com/cosmos/interchain-security/x/ccv/types"
+	abci "github.com/tendermint/tendermint/abci/types"
+
+	"github.com/stretchr/testify/require"
+)
+
+// TestQueueVSCs tests queueing validator set updates.
+func TestQueueValidatorUpdates(t *testing.T) {
+	key := ibcsimapp.CreateTestPubKeys(1)[0]
+	tmPubKey, _ := cryptocodec.ToTmProtoPublicKey(key)
+
+	testCases := []struct {
+		name                     string
+		packets                  []ccv.ValidatorSetChangePacketData
+		expectNextValsetUpdateId uint64
+		expectedQueueSize        int
+	}{
+		{
+
+			name:                     "no updates to send",
+			packets:                  []ccv.ValidatorSetChangePacketData{},
+			expectNextValsetUpdateId: 1,
+			expectedQueueSize:        0,
+		},
+		{
+			name: "have updates to send",
+			packets: []ccv.ValidatorSetChangePacketData{
+				{
+					ValidatorUpdates: []abci.ValidatorUpdate{
+						{PubKey: tmPubKey, Power: 1},
+					},
+					ValsetUpdateId: 1,
+				},
+			},
+			expectNextValsetUpdateId: 1,
+			expectedQueueSize:        1,
+		},
+	}
+
+	chainID := "consumer"
+
+	for _, tc := range testCases {
+		keeperParams := testkeeper.NewInMemKeeperParams(t)
```

Comment:
This area can be simplified unless you're doing custom param setup
```suggestion
		pk, ctx, ctrl, mocks := testkeeper.GetProviderKeeperAndCtx(t, testkeeper.NewInMemKeeperParams(t))
```